src/core/reactor.cc

```diff
@@ -662,8 +662,8 @@ reactor::signals::signal_handler::signal_handler(int signo, noncopyable_function
 
 void
 reactor::signals::handle_signal(int signo, noncopyable_function<void ()>&& handler) {
-    _signal_handlers.emplace(std::piecewise_construct,
-        std::make_tuple(signo), std::make_tuple(signo, std::move(handler)));
+    signal_handler h(signo, std::move(handler));
+    _signal_handlers.insert_or_assign(signo, std::move(h));
```

Comment:
check the return value to skip the rest if performs assignmeng but not insertion?